src/common/file_system/virtual_file_system.cpp

```diff
@@ -0,0 +1,85 @@
+#include "common/file_system/virtual_file_system.h"
+
+#include "common/file_system/local_file_system.h"
+
+namespace kuzu {
+namespace common {
+
+VirtualFileSystem::VirtualFileSystem() {
+    defaultFS = std::make_unique<LocalFileSystem>();
+}
+
+FileSystem* VirtualFileSystem::findFileSystem(const std::string& /*path*/) {
+    return defaultFS.get();
+}
+
+std::unique_ptr<FileInfo> VirtualFileSystem::openFile(
+    const std::string& path, int flags, FileLockType lockType) {
+    return findFileSystem(path)->openFile(path, flags, lockType);
+}
+
+void VirtualFileSystem::readFromFile(
```

Comment:
As discussed, we shouldn't implement `readFromFile`, `readFile`, `writeFile`, and `seek` for VirtualFileSystem for now.